mlflow/utils/search_utils.py

```diff
@@ -488,5 +490,26 @@ def parse_filter_for_model_registry(cls, filter_string):
             raise MlflowException("Invalid clause(s) in filter string: %s. "
                                   "%s" % (invalid_clauses, expected),
                                   error_code=INVALID_PARAMETER_VALUE)
-        return [cls._get_comparison_for_model_registry(si)
-                for si in statement.tokens if isinstance(si, Comparison)]
+        return statement
+
+    @classmethod
+    def parse_filter_for_model_versions(cls, filter_string):
+        statement = cls._parse_filter_for_model_registry(filter_string)
+        if statement:
+            return [cls._get_comparison_for_model_registry(
+                si,
+                cls.VALID_SEARCH_KEYS_FOR_MODEL_VERSIONS)
+                    for si in statement.tokens if isinstance(si, Comparison)]
+        else:
+            return []
```

Comment:
why not push `cls.VALID_SEARCH_KEYS_FOR_MODEL_VERSIONS` into `_parse_filter_for_model_registry` method and [DRY this code out](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself), like

```
@classmethod
def _parse_filter_for_model_registry(cls, filter_string, valid_search_keys):
  # ...
  # existing unchanged code
  # ...
  return [cls._get_comparison_for_model_registry(si, valid_search_keys)
          for si in statement.tokens if isinstance(si, Comparison)]	  

@classmethod
def parse_filter_for_model_versions(cls, filter_string):
  return cls._parse_filter_for_model_registry(filter_string, cls.VALID_SEARCH_KEYS_FOR_MODEL_VERSIONS)

@classmethod
def parse_filter_for_registered_models(cls, filter_string):
  return cls._parse_filter_for_model_registry(filter_string, cls.VALID_SEARCH_KEYS_FOR_REGISTERED_MODELS)
```

Reply:
Yeah, this makes sense - thanks for the suggestion!